python/ql/src/experimental/Security/CWE-295/UrlopenWithoutValidation.ql

```diff
@@ -0,0 +1,45 @@
+/**
+ * @name Attempt to open url without certificate validation
+ * @description Opening a url without certificate validation can allow man-in-the-middle attacks.
+ * @kind problem
+ * @problem.severity error
+ * @id py/urlopen-without-cert-validation
+ * @tags security
+ *       external/cwe/cwe-295
+ */
+
+import python
+import semmle.python.ApiGraphs
+
+predicate calls_urllib(CallNode call) {
+  call =
+    API::moduleImport(["urllib", "urllib2"])
+        .getMember("request")
+        .getMember("urlopen")
+        .getACall()
+        .getNode()
+}
+
+predicate is_https(CallNode call) {
+  exists(ControlFlowNode http_arg, StringValue http_string |
+    http_arg = call.getArg(0) and
+    http_arg.pointsTo(http_string) and
+    http_string.getText().matches("https://%")
+  )
+}
+
+predicate contains_capath(CallNode call) {
+  not exists(ControlFlowNode verify | verify = call.getArgByName(["cafile", "capath", "context"]))
+  or
+  exists(ControlFlowNode empty |
+    empty = call.getArgByName(["cafile", "capath", "context"]) and
+    empty = API::builtin("None").getAnImmediateUse().asCfgNode()
+  )
+}
```

Comment:
```suggestion
predicate validates_certificate(DataFlow::CallCfgNode call) {
  exists(DataFlow::Node arg |
    arg = call.getArgByName(["cafile", "capath", "context"]) and
    not arg.getALocalSource().asExpr() instanceof None
  )
}
```
As before, we use `getALocalSource` to track the argument back to its local source. This should be slightly more general than `API::builtin("None").getAnImmediateUse()`, but still fairly precise.

I also think the name of the predicate should be more general. It's not checking for the presence of the `capath` argument alone.